### DISCUSSION TITLE
Linux version please?

### DISCUSSION THREAD

**Discussion by CHJ85** [project64/project64#1913] [Ideas]
Hi there. It'd be great if Proect64 was available for Linux.
There's Mupen64plus and Parallel, but these don't really have a good frontend GUI.
I could use something like RetroArch or EmulationStation. But even though they're great emulator frontends, they'll only import roms that are recognizable in the database. So that means no custom rom hacks essentially.

**Comment by flagrama** [comment]
ParaLLEl is also just mupen64plus.

RetroArch may not add the unknown files to the playlist, but you can still load them just fine from the file browser.

Project 64's core is platform-independent so you can build the core (with a little tweaking currently) but there is no frontend to handle it currently. [As you can see here](https://www.pj64-emu.com/support-project64.html), full multiplatform support is a ways off planning-wise, but it is something actually being thought about. It's just that other things are currently more important.

**Comment by CHJ85** [comment]
Thanks for your quick response.
I like the Project64 gui because it has everything.
Would you be willing to develop the PJ64 frontend as a Mupen64Plus gui for Linux? Idk if that would save you some time compared to rebuilding the entire program.

**Comment by Squall-Leonhart** [comment]
no

**Comment by CHJ85** [comment]
Yea I kinda figured.

**Comment by flagrama** [comment]
There is someone working on a mupen64plus frontend that looks extremely similar to Project 64 already.

**Comment by Jj0YzL5nvJ** [comment]
> Would you be willing to develop the PJ64 frontend as a Mupen64Plus gui for Linux?

Just like [RMG](https://emulation.gametechwiki.com/index.php/RMG)?

**Comment by project64** [comment]
feel free to continue to discuss, but github issue tracking is not the place for feature wishes

**Comment by CHJ85** [comment]
@Jj0YzL5nvJ Haven't heard about that project before.
Will have to look into it. Thanks.
@flagrama You mean RMG?

**Comment by flagrama** [comment]
> @flagrama You mean RMG?

Yes.

**Comment by CHJ85** [comment]
@flagrama Thanks. Although I could only find a Windows build.

**Reply by flagrama** [reply]
There are still no releases currently. You should be able to build it with cmake, probably Qt, and maybe the build dependencies of mupen64plus-core. You'll have to see for yourself by running `Source/Script/Build.sh`

**Comment by CHJ85** [comment]
@flagrama Right. Thank you! 👍🏽

**Comment by bdragoncore** [comment]
Maybe one day the UI for PJ64 will be based on Dear ImGUI: https://github.com/ocornut/imgui

This approach means PJ64 won't need to depend on frameworks like Qt, GTK or even GDI+, everything for the UI will only need to depend on the GPU just as the video plugins.
